### PR TITLE
Filling in the NJOY input string.Template

### LINKED ISSUE #54: Fill NJOY/GROUPR template for a given nuclide and write file
Using the template defined in #49, complete it with necessary information for a given case (nuclide & reaction type) and write it to a file.

### PR DESCRIPTION
**Pull request by eitan-weinstein**
Closes #54.

Creates a method to substitute in material-specific values in the place of fillers in the static string.Template that stores the NJOY input file string.

### CODE REVIEW

File: src/DataLib/fendl32B_retrofit/groupr_tools.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    Z = str(elements[element]).zfill(2)
+    title = f'"{Z}-{element}-{A} for TENDL 2017"'
+
+    card9_lines = []
+    for i, MT in enumerate(MTs):
+        mtname = mt_dict[str(MT)]['Reaction']
+        card9_lines.append(f'{MFD} {MT} "{mtname}" /') 
+    card9 = '\n '.join(card9_lines)
+    
+    return njoy_template.substitute(
```
Comment (gonuke) [inline src/DataLib/fendl32B_retrofit/groupr_tools.py:94]: 👍

**Comment by gonuke** [comment]
Seems you have a lot of extra files here.... not sure where they came from...

**Comment by eitan-weinstein** [comment]
> Seems you have a lot of extra files here.... not sure where they came from...

Yeah, those accidentally came through when I was just rebasing. I should have them all removed now.

File: src/DataLib/fendl32B_retrofit/groupr_tools.py [hunk 2]
```diff
@@ -31,4 +31,80 @@
...
+    Returns:
+        template (string.Template): Modified template with the material-
+            specific information substituted in for the $identifiers.
+    """
+
+    Z = str(elements[element]).zfill(2)
+    title = f'"{Z}-{element}-{A} for TENDL 2017"'
+
+    card9_lines = []
+    for i, MT in enumerate(MTs):
```
Comment (gonuke) [inline src/DataLib/fendl32B_retrofit/groupr_tools.py:87]:
```suggestion
    for MT in MTs:
```

File: src/DataLib/fendl32B_retrofit/groupr_tools.py [hunk 3]
```diff
@@ -31,4 +31,80 @@
...
+        mtname = mt_dict[str(MT)]['Reaction']
```
Comment (gonuke) [inline src/DataLib/fendl32B_retrofit/groupr_tools.py:88]: Looking at how we use it, could we just use the `int` version of `MT` as the key to this dictionary all the way back to when you load it from the CSV?

File: src/DataLib/fendl32B_retrofit/process_fendl3.2.py [hunk 4]
```diff
@@ -8,5 +10,14 @@ def main():
 
     mt_dict = rxd.process_mt_data(rxd.load_mt_table('mt_table.csv'))
 
+    endf_path = 'files_for_tests/endf_test1.tendl'
+    pendf_path = 'files_for_tests/pendf_test1.pendf'
+
+    material_id, MTs = tpp.extract_endf_specs(endf_path)
+    njoy_template = groupr_tools.establish_static_template()
```
Comment (gonuke) [inline src/DataLib/fendl32B_retrofit/process_fendl3.2.py:17]: I don't see this function anymore?  Don't we just have to import the template?

**Review by gonuke — CHANGES_REQUESTED** [review]
A few little comments here

File: src/DataLib/fendl32B_retrofit/process_fendl3.2.py (hunk 4 above)
Comment (eitan-weinstein) [reply src/DataLib/fendl32B_retrofit/process_fendl3.2.py:17]: Yep, you're right. That was a holdover from a previous version. I'll take that out.

File: src/DataLib/fendl32B_retrofit/groupr_tools.py (hunk 3 above)
Comment (eitan-weinstein) [reply src/DataLib/fendl32B_retrofit/groupr_tools.py:88]: Sure, that makes sense.

**Comment by eitan-weinstein** [comment]
As it stands, I currently have an example usage in the `main()` function for `process_fendl3.2.py`. Is that something you want in this PR or should I pull it out?

**Review by gonuke — CHANGES_REQUESTED** [review]
This is good, and the demonstration in `main()` is great.  My last request here is to add a test and then I think we can merge.

File: src/DataLib/fendl32B_retrofit/groupr_tools.py [hunk 5]
```diff
@@ -31,4 +31,80 @@
...
+    'Pm', 'Sm', 'Eu', 'Gd', 'Tb', 'Dy', 'Ho', 'Er', 'Tm', 'Yb',
+    'Lu', 'Hf', 'Ta', 'W', 'Re', 'Os', 'Ir', 'Pt', 'Au', 'Hg',
+    'Tl', 'Pb', 'Bi', 'Po', 'At', 'Rn', 'Fr', 'Ra', 'Ac', 'Th',
+    'Pa', 'U', 'Np', 'Pu', 'Am', 'Cm', 'Bk', 'Cf', 'Es', 'Fm',
+    'Md', 'No', 'Lr', 'Rf', 'Db', 'Sg', 'Bh', 'Hs', 'Mt', 'Ds',
+    'Rg', 'Cn', 'Nh', 'Fl', 'Mc', 'Lv', 'Ts', 'Og'
+]
+elements = dict(zip(elements, range(1, len(elements)+1)))
+
+def fill_input_template(material_id, MTs, element, A, mt_dict):
```
[the remark this answers is not in the document]
Comment (eitan-weinstein) [reply src/DataLib/fendl32B_retrofit/groupr_tools.py:53]: Alright, I'll add a few tests in.

**Review by gonuke — APPROVED** [review]
Thanks @eitan-weinstein - this looks good.

**Pull request merged**